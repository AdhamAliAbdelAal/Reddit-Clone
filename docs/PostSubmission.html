<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: PostSubmission</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: PostSubmission</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>PostSubmission<span class="signature">()</span><span class="type-signature"> &rarr; {React.Component}</span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="PostSubmission"><span class="type-signature"></span>new PostSubmission<span class="signature">()</span><span class="type-signature"> &rarr; {React.Component}</span></h4>
    

    



<div class="description">
    <p>This component contains post andn save draft buttons</p>
</div>













    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>handleSaveDraft</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last"><p>Hanlding save draft button</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>handlePost</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last"><p>Hanlding post submitting</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>readyToPost</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last"><p>Check whether the post button is disabled or not</p></td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="CreatePost_PostSubmission_PostSubmission.jsx.html">CreatePost/PostSubmission/PostSubmission.jsx</a>, <a href="CreatePost_PostSubmission_PostSubmission.jsx.html#line14">line 14</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Container of buttons</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">React.Component</span>


    </dd>
</dl>

    





    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Communities.html">Communities</a></li><li><a href="CommunityItem.html">CommunityItem</a></li><li><a href="CommunityList.html">CommunityList</a></li><li><a href="CreatePostContainer.html">CreatePostContainer</a></li><li><a href="CreatePostForm.html">CreatePostForm</a></li><li><a href="CreatePostInHome.html">CreatePostInHome</a></li><li><a href="HomePageContainer.html">HomePageContainer</a></li><li><a href="HomePageFooter.html">HomePageFooter</a></li><li><a href="JoinButton.html">JoinButton</a></li><li><a href="MainContent.html">MainContent</a></li><li><a href="PersonalReddit.html">PersonalReddit</a></li><li><a href="Post.html">Post</a></li><li><a href="PostHeader.html">PostHeader</a></li><li><a href="PostList.html">PostList</a></li><li><a href="PostReactions.html">PostReactions</a></li><li><a href="PostSubmission.html">PostSubmission</a></li><li><a href="PostTags.html">PostTags</a></li><li><a href="PostsClassification.html">PostsClassification</a></li><li><a href="Reactions.html">Reactions</a></li><li><a href="RedditPremium.html">RedditPremium</a></li><li><a href="SideBar.html">SideBar</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 4.0.0</a> on Tue Nov 08 2022 16:27:42 GMT+0200 (Eastern European Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>