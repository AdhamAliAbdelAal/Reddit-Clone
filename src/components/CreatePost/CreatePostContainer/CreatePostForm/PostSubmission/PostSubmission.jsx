import { Box, Checkbox } from '@mui/material';
import RedditButton from '../../../../RedditButton/RedditButton';
import {
  ConnectAccounts, InfoIcon, PostButton, SendReplayFormLabel, SendReplyContainer,
} from './styles';
/**
 * This component contains post and save draft buttons
 *
 * @component PostHeader
 * @property {function} handleSaveDraft -Hanlding save draft button
 * @property {function} handlePost -Hanlding post submitting
 * @property {boolean} readyToPost -Check whether the post button is disabled or not
 * @property {boolean} sendReplies -Check whether the user need to be sent the post reply notifications or not.
 * @property {function} handleSendReplies -Hanlding post reply notifications
 * @returns {React.Component} Container of buttons
 */

function PostSubmission(props) {
  const {
    handleSaveDraft, handlePost, readyToPost, handleSendReplies, sendReplies,
  } = props;
  return (
    <Box>
      <Box m={2} gap={1} display="flex" justifyContent="flex-end">
        <RedditButton
          variant="outlined"
          padding="3px 16px"
          fontSize={14}
          fontWeight="bold"
          type="submit"
          onClick={handleSaveDraft}
        >
          save draft
        </RedditButton>
        <PostButton
          variant="contained"
          type="submit"
          onClick={handlePost}
          disabled={!readyToPost}
          data-testid="post button"
        >
          post
        </PostButton>
      </Box>
      <SendReplyContainer p={2}>
        <SendReplayFormLabel
          control={(
            <Checkbox
              onChange={handleSendReplies}
              checked={sendReplies}
              size="small"
              disableRipple
            />
)}
          label="Send me post reply notifications"
        />
        <Box display="flex">
          <ConnectAccounts>
            Connect accounts to share your post
          </ConnectAccounts>
          <InfoIcon />
        </Box>
      </SendReplyContainer>
    </Box>
  );
}

export default PostSubmission;
