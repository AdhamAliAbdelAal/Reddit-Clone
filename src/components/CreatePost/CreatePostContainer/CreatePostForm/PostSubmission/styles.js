import { Box, FormControlLabel } from '@mui/material';
import { styled } from '@mui/system';
import { Link } from 'react-router-dom';
import { AiOutlineInfoCircle } from 'react-icons/ai';
import RedditButton from '../../../../RedditButton/RedditButton';

export const PostButton = styled(RedditButton)(() => ({
  padding: '3px 16px',
  fontSize: 14,
  fontWeight: 'bold',
  '&.Mui-disabled': {
    cursor: 'not-allowed',
    pointerEvents: 'auto',
    backgroundColor: '#8D8D8D',
    color: '#ffffff80',
  },
}));

export const SendReplyContainer = styled(Box)(() => ({
  backgroundColor: '#f6f7f8',
}));

export const SendReplayFormLabel = styled(FormControlLabel)(() => ({
  margin: '5px 0 10px',
  '& .MuiFormControlLabel-label': {
    fontSize: 14,
    lineHeight: '18px',
    color: '#000',
    fontWeight: 500,

  },
  '& .MuiCheckbox-root': {
    padding: 0,
    marginRight: 5,
  },
  '& .MuiSvgIcon-root': {
    fontSize: 20,
  },
}));

export const ConnectAccounts = styled(Link)(({ theme }) => ({
  fontSize: 14,
  lineHeight: '18px',
  fontWeight: 500,
  color: theme.palette.primary.main,
  paddingLeft: 1,
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'underline',
  },
}));

export const InfoIcon = styled(AiOutlineInfoCircle)(({ theme }) => ({
  width: 21,
  height: 22,
  transform: 'translateY(-15%)',
  color: theme.palette.third.main,
  marginLeft: 5,
}));
