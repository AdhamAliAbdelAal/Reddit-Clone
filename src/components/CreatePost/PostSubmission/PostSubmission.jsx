import { Box } from '@mui/material';
import RedditButton from '../../RedditButton/RedditButton';

function PostSubmission(props) {
  const { handleSaveDraft, handlePost } = props;
  return (
    <Box m={2} gap={1} display="flex" justifyContent="flex-end">
      <RedditButton
        variant="outlined"
        padding="3px 16px"
        fontSize={14}
        fontWeight="bold"
        type="submit"
        onClick={handleSaveDraft}
      >
        save draft
      </RedditButton>
      <RedditButton
        variant="contained"
        padding="3px 16px"
        fontSize={14}
        fontWeight="bold"
        type="submit"
        onClick={handlePost}
      >
        post
      </RedditButton>
    </Box>
  );
}

export default PostSubmission;
