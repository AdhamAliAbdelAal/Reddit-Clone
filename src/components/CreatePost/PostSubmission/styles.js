import { styled } from '@mui/system';
import RedditButton from '../../RedditButton/RedditButton';

export const PostButton = styled(RedditButton)(() => ({
  padding: '3px 16px',
  fontSize: 14,
  fontWeight: 'bold',
  '&.Mui-disabled': {
    cursor: 'not-allowed',
    pointerEvents: 'auto',
  },
}));
