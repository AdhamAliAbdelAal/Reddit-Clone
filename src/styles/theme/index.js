import { createTheme } from '@mui/material';

const theme = createTheme({
  palette: {
    primary: {
      main: '#0079D3',
      light: '#1484D6',
      dark: '#1484D6',
      fade: '#3394DC',
    },
    secondary: {
      main: '#FF4500',
      light: '#FF5414',
      dark: '#FF5414',
    },
    third: {
      main: '#898989',
      light: '#E8E8E8',
      dark: '#E8E8E8',
    },
    action: {
      hoverOpacity: 0.1,
    },
    neutral: {
      main: '#978A8C',
      contrastText: '#fff',
    },
    white: {
      main: '#ffffff',
      light: '#ffffff',
      dark: '#898989',
    },
    offwhite: {
      main: '#DAE0E6',
      dark: '#2E2F30',
    },
  },
});

export const fonts = {
  'system-ui': 'system-ui',
};
export default theme;
