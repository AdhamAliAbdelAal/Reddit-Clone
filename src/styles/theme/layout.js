import { createTheme } from '@mui/material';

const layoutTheme = (theme) => createTheme({
  ...theme,
  breakpoints: {
    values: {
      xxs: 335,
      xs: 420,
      sm: 600,
      md: 900,
      lg: 1200,
      xl: 1536,
    },
  },
});
export default layoutTheme;
