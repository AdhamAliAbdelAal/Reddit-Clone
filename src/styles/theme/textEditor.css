.rdw-option-wrapper {
    background-color: #f6f7f8 !important;
    padding: 5px 3px !important;
    border-radius: 7px !important;
    border-width: 0 !important;
    margin: 0 3px !important;
  }
  .rdw-option-wrapper:hover,.rdw-option-active{
    background-color: #1a1a1b1a !important;
    box-shadow: none !important;
  }

.public-DraftStyleDefault-block{
    margin: 0 !important;
    margin-top: 5px !important;
}