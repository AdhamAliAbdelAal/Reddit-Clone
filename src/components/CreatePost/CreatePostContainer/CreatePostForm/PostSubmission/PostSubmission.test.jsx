import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { ThemeProvider } from '@mui/system';
import { MemoryRouter as Router } from 'react-router-dom';
import PostSubmission from './PostSubmission';
import theme from '../../../../../styles/theme';

describe(PostSubmission, () => {
  const handleSaveDraft = jest.fn((param) => param);
  const handlePost = jest.fn((param) => param);

  afterEach(() => {
    handleSaveDraft.mockReset();
    handlePost.mockReset();
  });

  it('PostSubmission renders correctly', () => {
    const testCase = {
      handleSaveDraft, handlePost, readyToPost: true,
    };
    const tree = renderer
      .create(
        <Router>
          <ThemeProvider theme={theme}>
            <PostSubmission
              handleSaveDraft={testCase.handleSaveDraft}
              handlePost={testCase.handlePost}
              readyToPost={testCase.readyToPost}
            />
          </ThemeProvider>
        </Router>
        ,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('PostSubmission posts correctly', () => {
    const testCase = {
      handleSaveDraft, handlePost, readyToPost: true,
    };
    render(
      <Router>
        <ThemeProvider theme={theme}>
          <PostSubmission
            handleSaveDraft={testCase.handleSaveDraft}
            handlePost={testCase.handlePost}
            readyToPost={testCase.readyToPost}
          />
        </ThemeProvider>
      </Router>,
    );
    const postButton = screen.getByTestId('post button');
    expect(handlePost.mock.calls.length).toEqual(0);
    fireEvent.click(postButton);
    expect(handlePost.mock.calls.length).toEqual(1);
    fireEvent.click(postButton);
    expect(handlePost.mock.calls.length).toEqual(2);
  });

  it('Post button is disabled correctly', () => {
    const testCase = {
      handleSaveDraft, handlePost, readyToPost: false,
    };
    render(
      <Router>
        <ThemeProvider theme={theme}>
          <PostSubmission
            handleSaveDraft={testCase.handleSaveDraft}
            handlePost={testCase.handlePost}
            readyToPost={testCase.readyToPost}
          />
        </ThemeProvider>
      </Router>,
    );
    const postButton = screen.getByTestId('post button');
    expect(handlePost.mock.calls.length).toEqual(0);
    fireEvent.click(postButton);
    expect(handlePost.mock.calls.length).toEqual(0);
  });
});
