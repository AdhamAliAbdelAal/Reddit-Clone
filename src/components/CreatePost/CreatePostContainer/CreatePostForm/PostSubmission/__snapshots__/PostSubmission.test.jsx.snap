// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSubmission PostSubmission renders correctly 1`] = `
<div
  className="MuiBox-root css-1ra4eh8"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1g9iwm8-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    fontSize={14}
    fontWeight="bold"
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    padding="3px 16px"
    tabIndex={0}
    type="submit"
  >
    save draft
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-bu9uc5-MuiButtonBase-root-MuiButton-root"
    data-testid="post button"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    post
  </button>
</div>
`;
